core/src/common.cpp

```diff
@@ -362,6 +364,434 @@ namespace irods::http
+				}
+				else if (algorithm_specifics == "512") {
+					logging::trace("{}: Adding [{}] to allowed verification algorithms.", __func__, _alg);
+					_verifier.allow_algorithm(jwt::algorithm::rs512(pub_key));
+				}
+			}
+			// "PS"
+			else {
+				if (algorithm_specifics == "256") {
+					logging::trace("{}: Adding [{}] to allowed verification algorithms.", __func__, _alg);
+					_verifier.allow_algorithm(jwt::algorithm::ps256(pub_key));
+				}
+				else if (algorithm_specifics == "384") {
+					logging::trace("{}: Adding [{}] to allowed verification algorithms.", __func__, _alg);
+					_verifier.allow_algorithm(jwt::algorithm::ps384(pub_key));
+				}
+				else if (algorithm_specifics == "512") {
+					logging::trace("{}: Adding [{}] to allowed verification algorithms.", __func__, _alg);
+					_verifier.allow_algorithm(jwt::algorithm::ps512(pub_key));
+				}
+			}
+			return;
+		}
+		else if (algorithm_family == "ES") {
+			logging::trace("{}: Detected [ES], attempting extraction of attributes from JWK...", __func__);
+
+			// Get curve parameter (JWA Section 6.2.1)
+			auto crv{_jwk.get_curve()};
+
+			// Get x coordinate parameter (JWA Section 6.2.1)
+			auto x{_jwk.get_jwk_claim("x").as_string()};
+
+			// Get y coordinate parameter (JWA Section 6.2.1)
+			// MUST be present if 'crv' is 'P-256', 'P-384', or 'P-521' (JWA Section 6.2.1)
+			auto y{_jwk.get_jwk_claim("y").as_string()};
+
+			// Create public key
+			auto pub_key{jwt::helper::create_public_key_from_ec_components(crv, x, y)};
+
+			// Add verification algorithm
+			if (algorithm_specifics == "256") {
+				logging::trace("{}: Adding [{}] to allowed verification algorithms.", __func__, _alg);
+				_verifier.allow_algorithm(jwt::algorithm::es256(pub_key));
+			}
+			else if (algorithm_specifics == "384") {
+				logging::trace("{}: Adding [{}] to allowed verification algorithms.", __func__, _alg);
+				_verifier.allow_algorithm(jwt::algorithm::es384(pub_key));
+			}
+			else if (algorithm_specifics == "512") {
+				logging::trace("{}: Adding [{}] to allowed verification algorithms.", __func__, _alg);
+				_verifier.allow_algorithm(jwt::algorithm::es512(pub_key));
+			}
+			return;
+		}
+
+		logging::warn("{}: Algorithm [{}] is not supported.", __func__, _alg);
+	}
+
+	/// Adds verification algorithm(s) to the \p _verifier based on either the given 'kid' or algorthims matching
+	/// the family of the desired algorithm \p _alg.
+	///
+	/// See RFC 7518 for details on JSON Web Algorithms (JWA)
+	///
+	/// \param[in,out] _verifier The jwt::verifier to add additional verification algorithms to.
+	/// \param[in]     _jwks     The JWKs to search through.
+	/// \param[in]     _jwt      The decoded JWT that needs to be verified.
+	///
+	/// \returns A refernce to the provided jwt::verifier, \p _verifier, allowing for chaining.
+	auto add_algorithms_to_verifier(
+		jwt::verifier<jwt::default_clock, jwt::traits::nlohmann_json>& _verifier,
```

Comment:
Would a type alias be appropriate for this since it has popped up in many places?

Reply:
I'm terrible at names. `nlohmann_json_verifier`? Just `jwt_verifier`?

Suggestions appreciated.

Reply:
I personally think `jwt_verifier` is fine unless we plan on supporting / implementing other kinds of `jwt::verifier` types in the future. What you have now is also just fine.

Reply:
Done.